java/src/jmri/jmrit/roster/swing/speedprofile/SpeedProfilePanel.java

```diff
@@ -793,5 +996,216 @@ float getReverseSpeed() {
+                @SuppressWarnings("unchecked")
+                List<Element> l = root.getChild("sensors").getChildren("sensor");
+                if (log.isDebugEnabled()) {
+                    log.debug("readFile sees " + l.size() + " sensors");
+                }
+                for (int i = 0; i < l.size(); i++) {
+                    Element e = l.get(i);
+                    String sensorType = e.getChild("sensorname").getText();
+                    switch (sensorType) {
+                        case "sensorAPanel": sensorAPanel.setDefaultNamedBean(
+                                InstanceManager.sensorManagerInstance().getByUserName(e.getChild("sensorvalue").getText()));
+                            break;
+                        case "sensorBPanel": sensorBPanel.setDefaultNamedBean(
+                                InstanceManager.sensorManagerInstance().getByUserName(e.getChild("sensorvalue").getText()));
+                            break;
+                        case "sensorCPanel": sensorCPanel.setDefaultNamedBean(
+                                InstanceManager.sensorManagerInstance().getByUserName(e.getChild("sensorvalue").getText()));
+                            break;
+                        default:
+                            log.warn("Invalid Sensor found in DecoderProSpeedProfile.xml");
+                    }
+                }
+            }
+            if (root.getChild("steps") != null) {
+                @SuppressWarnings("unchecked")
+                List<Element> l = root.getChild("steps").getChildren();
+                for (int i = 0; i < l.size(); i++) {
+                    Element e = l.get(i);
+                    switch (e.getName()) {
+                    case "speedStepFrom": speedStepFrom.setText(e.getValue());
+                        break;
+                    case "speedStepTo": speedStepTo.setText(e.getValue());
+                        break;
+                    case "speedStepIncr": speedStepIncr.setText(e.getValue());
+                        break;
+                    default:
+                        log.warn("Invalid field in steps of PanelProSpeedProfiler.xml");
+                    }
+                }
+            }
+
+        } catch (JDOMException ex) {
+            log.error("File invalid: " + ex);
+        } catch (IOException ex) {
+            log.error("Error reading file: " + ex);
+        }
+
+        log.debug("...done");
+    }
+
+    private static class SpeedProfilerXml extends XmlFile {
+
+        public static String getDefaultFileName() {
+            return getFileLocation() + getFileName();
+        }
+
+        public File getFile(boolean store) {
+            File file = findFile(getDefaultFileName());
+            if (file == null && store) {
+                file = new File(getDefaultFileName());
+            }
+            return file;
+        }
+
+        private static String baseFileName = "SpeedProfiler.xml";
+
+        public static String getFileName() {
```

Comment:
Following up on @rhwood's comment, it would be better if you used the "portable filename" methods for pathnames to files.  That way you can store things in a common place, etc.  There's  a [rudimentary page on it](http://jmri.org/help/en/html/doc/Technical/FileNames.shtml) in the JMRI developer pages at [http://jmri.org/help/en/html/doc/Technical/FileNames.shtml](http://jmri.org/help/en/html/doc/Technical/FileNames.shtml).